storage/src/store/base.rs

```diff
@@ -0,0 +1,845 @@
+        }
+
+        let op = Fixed::Update(key, value);
+        self.apply_op(op).await?;
+
+        Ok(res)
+    }
+
+    /// Delete the given key and its associated value.
+    ///
+    /// If the key already has no value, the operation is treated as a no-op and no log entry is created.
+    ///
+    /// The deletion is immediately visible in subsequent queries, but remains uncommitted until
+    /// [`Store::commit`] is called. Uncommitted deletions will be rolled back if the store is
+    /// closed without committing.
+    ///
+    /// # Arguments
+    ///
+    /// * `key` - The key to delete
+    ///
+    /// # Returns
+    ///
+    /// * `Some(location)` - The log location of the deleted value, if the key existed
+    /// * `None` - The key had no value to delete
+    ///
+    /// # Errors
+    ///
+    /// Returns [`Error`] if:
+    /// - The delete operation cannot be written to the log
+    /// - The snapshot index cannot be updated
+    pub async fn delete(&mut self, key: K) -> Result<Option<u64>, Error> {
+        let loc = self.op_count();
+        let r = Self::delete_key(&mut self.snapshot, &self.log, &key, loc).await?;
+        if r.is_some() {
+            self.apply_op(Fixed::Deleted(key)).await?;
+        };
+
+        Ok(r)
+    }
+
+    /// Commit all pending operations to persistent storage.
+    ///
+    /// This method ensures that all operations performed since the last commit are durably
+    /// persisted and will be recovered if the store is reopened. The commit process involves:
+    ///
+    /// 1. Advancing the inactivity floor to mark superseded operations as inactive
+    /// 2. Writing a commit marker to the operation log
+    /// 3. Synchronizing all data to persistent storage
+    /// 4. Pruning inactive operations to reclaim space
+    ///
+    /// After a successful commit, the uncommitted operation count is reset to zero.
+    ///
+    /// # Errors
+    ///
+    /// Returns [`Error`] if:
+    /// - The commit marker cannot be written to the log
+    /// - Data cannot be synchronized to persistent storage
+    /// - Pruning of inactive operations fails
+    ///
+    /// # Note
+    ///
+    /// This operation may be expensive as it involves disk I/O and potentially significant
+    /// computation to identify and prune inactive operations.
+    pub async fn commit(&mut self) -> Result<(), Error> {
+        // Raise the inactivity floor by the # of uncommitted operations, plus 1 to account for the
+        // commit op that will be appended.
+        self.raise_inactivity_floor(self.uncommitted_ops + 1)
+            .await?;
+        self.uncommitted_ops = 0;
+        self.sync().await?;
+
+        // TODO: Make the frequency with which we prune known inactive items configurable in case
+        // this turns out to be a significant part of commit overhead, or the user wants to ensure
+        // the log is backed up externally before discarding.
+        self.prune_inactive().await
```

Comment:
In the office, we discussed making the actual `prune` call explicit (rather than running automatically).

Because we aren't syncing anything from `store`, this is ok but we'd need to offer some additional configuration if we want to integrate into `adb::any` (as we need to keep some before the inactive boundary to support dynamic state sync).